core/src/metadata/metadata.cc

```
@@ -33,9 +33,8 @@
 #include "metadata.h"
 #include <cassert>
 #include <cstring>
-#include <openssl/md5.h>
-
 
+#include "tiledb_openssl_shim.h"
```

tiledb_openssl_shim.h has been included, but there seem to be no openssl3 related changes in this file. Why?

---

Here also we are calling MD5() function at line 104 & 417, so we need tiledb_openssl_shim.h file

---

Same as my previous comment above. Sorry for the nitpicking,

---

Here I get the error as:
/nfs/home/santhosh/workbenchKms/2tdb_ossl3_aws_gcs/TileDB/core/src/metadata/metadata.cc: In member function ‘int Metadata::read(const char*, void**, size_t*)’:
/nfs/home/santhosh/workbenchKms/2tdb_ossl3_aws_gcs/TileDB/core/src/metadata/metadata.cc:102:3: error: ‘MD5’ was not declared in this scope
  102 |   MD5((const unsigned char*) key, strlen(key)+1, (unsigned char*) coords);
      |   ^~~
/nfs/home/santhosh/workbenchKms/2tdb_ossl3_aws_gcs/TileDB/core/src/metadata/metadata.cc: In member function ‘void Metadata::compute_array_coords(const char*, size_t, void*&, size_t&) const’:
/nfs/home/santhosh/workbenchKms/2tdb_ossl3_aws_gcs/TileDB/core/src/metadata/metadata.cc:415:5: error: ‘MD5’ was not declared in this scope
  415 |     MD5(keys_c, key_size, coords_c);
      |     ^~~
make[2]: *** [core/CMakeFiles/TILEDB_CORE_OBJECTS.dir/build.make:412: core/CMakeFiles/TILEDB_CORE_OBJECTS.dir/src/metadata/metadata.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:571: core/CMakeFiles/TILEDB_CORE_OBJECTS.dir/all] Error 2
make: *** [Makefile:146: all] Error 2

---

One of the aims of this exercise was to not just suppress deprecation warnings but to start using the openssl3 functionality instead of openssl1. Using `tiledb_openssl_shim` only suppresses the error, we should also be using the appropriate MD5 api from openssl3. See https://www.openssl.org/docs/man3.1/man3/MD5.html for what should be used for replacements.